### PR TITLE
CP-41972: update hwinfo to support python3 ACK

### PR DESCRIPTION
**Pull request by alexhimmel**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: python
 env:
-    - TOXENV=py27
+    - TOXENV=py36
```
Comment (andyhhp) [inline .travis.yml]: Travis is well and truly dead.  Just delete this entire file.

File: hwinfo/pci/tests/test_lspci.py [hunk 2]
```diff
@@ -43,12 +43,12 @@ def test_pci_device_class_name(self):
 
     def test_pci_device_sub_string(self):
         rec = self.parser.parse_items().pop()
-	self._assert_rec_key(rec, 'pci_device_sub_string')
-
+        self._assert_rec_key(rec, 'pci_device_sub_string')
+
```
Comment (andyhhp) [inline hwinfo/pci/tests/test_lspci.py]: You appear to have introduced trailing whitespace here, and in the next hunk too.

File: hwinfo/tools/inspector.py [hunk 3]
```diff
@@ -170,6 +170,8 @@ def get_info(self):
 
     def get_cpu_info(self):
         data = self.get_cpuinfo_data()
+        if data.startswith("b"):
+            data = data[2:]
```
Comment (andyhhp) [inline hwinfo/tools/inspector.py]: This doesn't look like it's related to a Py2->3 conversion.

**Comment by andyhhp** [comment]
So this latest commit looks fine.  However, don't we want to retain at least Py2 compatibility?  Surely we'll be running a newer ACK on existing the existing 8.x product line for the foreseeable future?

You might be able to get away with nothing more than having `from __future__ import print_function` on the scripts which implement main

**Comment by tianxiahcp** [comment]
Hi @andyhhp 
Handing strings in py2 and py3 is different.

- In py2 the str and bytes are similar

- In py3 the str and bytes are not similar, str is like the py2 unicode type, It is possible to encode() text to binary data, or decode() bytes into a text string, using a particular encoding.
I am not sure if can retain  Py2 compatibility, ACK python3 will support for rt-next

**Review by DeliZhangX — APPROVED** [review]

**Comment by MarkSymsCtx** [comment]
> Hi @andyhhp Handing strings in py2 and py3 is different.
> 
> * In py2 the str and bytes are similar
> * In py3 the str and bytes are not similar, str is like the py2 unicode type, It is possible to encode() text to binary data, or decode() bytes into a text string, using a particular encoding.
>   I am not sure if can retain  Py2 compatibility, ACK python3 will support for rt-next

You can, by means of using future or six (not a combination of them) and suitable imports of compatibility modules.

**Review by catherineliuchina — APPROVED** [review]

**Comment by alexhimmel** [comment]
Hi @MarkSymsCtx 
Thank you for your suggestions. 
ACK python2 only used for yangtze, if build ACK ISO for yangtze will use python-hwinfo py2, otherwise will use python-hwinfo py3, it seems no need to retain py2 compatibility. 
CC: @andyhhp

**Pull request merged**